andela/odin-ah-backend
# #160452465 - Setup Postgresql Database

#### What does this PR do?
Setup PostgreSQL as DB Engine
#### Description of Task to be completed?
- Installed and initialized Sequelize npm package with Postgres support
- Completely replaced mongoose with Sequelize
- Integrate Sequelize migration.
- Refactored user route functions into UserController
#### How should this be manually tested?
* Before installing, ensure [PostgreSQL](https://www.postgresql.org/docs/9.4/static/tutorial-install.html) is installed and you've create a database.
* From the terminal, run the following commands to clone and install application dependencies. 
``` 
> git clone https://github.com/andela/odin-ah-backend
> cd odin-ah-backend
> git checkout ch/160452465/setup-postgresql-database
> npm install
``` 
* Rename `.env-sample` file on the root directory of the clone repo to `.env`. Uncomment and supply database connection information.

```$xslt
> mv .env-sample .env
```
```
# DB_USERNAME=
# DB_PASSWORD=
# DB_NAME=
# DB_HOST=
# DB_PORT=
```
* Run Sequelize migration 
```$xslt
 > npm run migration
 ```

* To run application
 
 ```$xslt
 > npm start
 ```

* Using a REST client such as postman to test endpoints: `POST /api/users` and `PUT /api/user`.
sample request body
```
  {
      "email": "tmorolari@gmail.com",
      "username": "macphilips",
      "password": "password"
  }
```

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- [#160452465](<https://www.pivotaltracker.com/story/show/160452465>)
#### Screenshots (if appropriate)
- None
#### Questions:
